src/tty-ask-password-agent/tty-ask-password-agent.c

```diff
@@ -212,6 +231,80 @@ static int ask_password_plymouth(
         return 0;
 }
 
+static void free_consoles(struct console *con, unsigned int num) {
+        unsigned int n;
+
+        if (!con || num == 0)
+                return;
+
+        for (n = 0; n < num; n++)
+                free(con[n].tty);
+
+        free(con);
+}
+
+static int collect_consoles(struct console **consoles, unsigned int *num) {
+        _cleanup_free_ char *active = NULL;
+        const char *word, *state;
+        struct console *con = NULL;
+        size_t con_len = 0, len;
+        unsigned int count = 0;
+        int ret;
+
+        assert(num);
+        assert(consoles);
+
+        ret = read_one_line_file("/sys/class/tty/console/active", &active);
+        if (ret < 0)
+                return log_error_errno(ret, "Failed to read /sys/class/tty/console/active: %m");
+
+        FOREACH_WORD(word, len, active, state) {
```

Comment:
AFAICS the macro FOREACH_WORD is not deprecated but used overall ... and why is the len argument is a messy one?  The number of words in "/sys/class/tty/console/active" can not be more then 8 and the separator is a normal ASCII space.  This is not the kernels command line.

I'm going to miss the art of hardware dependent programming in real-time on big endian with extrem small memory and cache lines :)   ... Do you seriously think the huge footprint extract_first_word() from src/basic/util.c has any advantage as replacement to the small foorprint of the FOREACH_WORD() macro?